v2/observer/jobs.go

```diff
@@ -176,3 +191,64 @@ func (o *observer) deleteJobResources(
 		)
 	}
 }
+
+func (o *observer) startJobPodTimer(ctx context.Context, pod *corev1.Pod) {
+	if pod.Status.Phase != corev1.PodPending &&
+		pod.Status.Phase != corev1.PodRunning {
+		return
+	}
+	if pod.Annotations["timeoutDuration"] == "" {
+		return
+	}
+
+	duration := pod.Annotations["timeoutDuration"]
+	timeout, err := time.ParseDuration(duration)
+	if err != nil {
+		o.errFn(
+			errors.Wrapf(
+				err,
+				"unable to parse timeout duration %q for pod %q",
+				duration,
+				pod.Name,
+			),
+		)
+		return
+	}
+
+	go func() {
+		ticker := time.NewTimer(timeout)
+		defer ticker.Stop()
+
+		select {
+		case <-ticker.C:
+			status := core.JobStatus{
+				Phase: core.JobPhaseTimedOut,
+			}
+			// Use the API to update Job status
+			eventID := pod.Labels[myk8s.LabelEvent]
+			jobName := pod.Labels[myk8s.LabelJob]
+			apiRequestCtx, cancel := context.WithTimeout(
+				context.Background(),
+				apiRequestTimeout,
+			)
+			defer cancel()
+			if err := o.updateJobStatusFn(
+				apiRequestCtx,
+				eventID,
+				jobName,
+				status,
+			); err != nil {
+				o.errFn(
+					errors.Wrapf(
+						err,
+						"error updating status for event %q worker job %q",
+						eventID,
+						jobName,
+					),
+				)
+			}
+			go o.deleteJobResourcesFn(pod.Namespace, pod.Name, eventID, jobName)
```

Comment:
Instead of updating the status visa API and then going directly to the substrate to delete job-related resources, imho, it would be preferable to take the same "API is the brains of this outfit" route that we have taken elsewhere-- invoke a new "timeout job" operation on the jobs client and let the API do all the work.

This is why `TIME_OUT` was initially missing from the [job status update schema](https://github.com/brigadecore/brigade/pull/1411/files#r639028280). It was left out with the assumption that the observer doesn't directly _observe_ a pod that has timed out and then have a need to report that status, rather it was assumed the observer would observe that time has elapsed for a running pod and can _request_ the API server to execute the timeout.

Reply:
Ah, I see.  Sounds good.  I will transition this logic into a new API client func/operation.